src/types/typeGuards/single.ts

```diff
@@ -0,0 +1,813 @@
+ *
+ * @category Types - Type Guards
+ * @example
+ * ```ts
+ * declare const type: ts.Type;
+ *
+ * if (isFalseLiteralType(type)) {
+ *   // ...
+ * }
+ * ```
+ */
+export function isFalseLiteralType(type: ts.Type): type is FalseLiteralType {
+	return isBooleanLiteralType(type) && type.intrinsicName === "false";
+}
+
+/**
+ * Test if a type is a `FreshableType`.
+ *
+ * @category Types - Type Guards
+ * @example
+ * ```ts
+ * declare const type: ts.Type;
+ *
+ * if (isFreshableType(type)) {
+ *   // ...
+ * }
+ * ```
+ */
+export function isFreshableType(type: ts.Type): type is ts.FreshableType {
+	return isTypeFlagSet(type, ts.TypeFlags.Freshable);
+}
+
+/**
+ * Test if a type is a `IndexType`.
+ *
+ * @category Types - Type Guards
+ * @example
+ * ```ts
+ * declare const type: ts.Type;
+ *
+ * if (isIndexType(type)) {
+ *   // ...
+ * }
+ * ```
+ */
+export function isIndexType(type: ts.Type): type is ts.IndexType {
+	return isTypeFlagSet(type, ts.TypeFlags.Index);
+}
+
+/**
+ * Test if a type is a `IndexedAccessType`.
+ *
+ * @category Types - Type Guards
+ * @example
+ * ```ts
+ * declare const type: ts.Type;
+ *
+ * if (isIndexedAccessType(type)) {
+ *   // ...
+ * }
+ * ```
+ */
+export function isIndexedAccessType(
+	type: ts.Type
+): type is ts.IndexedAccessType {
+	return isTypeFlagSet(type, ts.TypeFlags.IndexedAccess);
+}
+
+/**
+ * Test if a type is a `InstantiableType`.
+ *
+ * @category Types - Type Guards
+ * @example
+ * ```ts
+ * declare const type: ts.Type;
+ *
+ * if (isInstantiableType(type)) {
+ *   // ...
+ * }
+ * ```
+ */
+export function isInstantiableType(type: ts.Type): type is ts.InstantiableType {
+	return isTypeFlagSet(type, ts.TypeFlags.Instantiable);
+}
+
+/**
+ * Test if a type is a `IntersectionType`.
+ *
+ * @category Types - Type Guards
+ * @example
+ * ```ts
+ * declare const type: ts.Type;
+ *
+ * if (isIntersectionType(type)) {
+ *   // ...
+ * }
+ * ```
+ */
+export function isIntersectionType(type: ts.Type): type is ts.IntersectionType {
+	return isTypeFlagSet(type, ts.TypeFlags.Intersection);
+}
+
+/**
+ * An intrinsic type.
+ *
+ * @category Type Types
+ */
+export interface IntrinsicType extends ts.Type {
+	intrinsicName: string;
+	objectFlags: ts.ObjectFlags;
+}
+
+/**
+ * Test if a type is an {@link IntrinsicType}.
+ *
+ * @category Types - Type Guards
+ * @example
+ * ```ts
+ * declare const type: ts.Type;
+ *
+ * if (isIntrinsicType(type)) {
+ *   // ...
+ * }
+ * ```
+ */
+export function isIntrinsicType(type: ts.Type): type is IntrinsicType {
+	// ts.TypeFlags.Intrinsic
+	const IntrinsicTypeFlags =
+		(ts.TypeFlags as { Intrinsic?: number }).Intrinsic ??
+		ts.TypeFlags.Any |
+			ts.TypeFlags.Unknown |
+			ts.TypeFlags.String |
+			ts.TypeFlags.Number |
+			ts.TypeFlags.BigInt |
+			ts.TypeFlags.Boolean |
+			ts.TypeFlags.BooleanLiteral |
+			ts.TypeFlags.ESSymbol |
+			ts.TypeFlags.Void |
+			ts.TypeFlags.Undefined |
+			ts.TypeFlags.Null |
+			ts.TypeFlags.Never |
+			ts.TypeFlags.NonPrimitive;
```

Comment:
[Performance] This doesn't need to be created each time the function is called, right? Assuming not: let's move it outside of the function?

Reply:
We could move this outside the function, but then we would be consuming more memory in order to always store this data, rather than just calculating it when it's needed. Those in either case the difference would be extremely minor. I'll make the change if you like.

Reply:
Yeah just storing a number is teeny. I'm still in favor :smile